src/Map/src/MapFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+use Symfony\UX\Map\Provider\ProviderInterface;
+
+/**
+ * Creates a map based on the configuration, and registers it in the map registry.
+ *
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+final readonly class MapFactory implements MapFactoryInterface
+{
+    public function __construct(
+        private ProviderInterface $defaultProvider,
+        private MapRegistry $mapRegistry,
+    ) {
+    }
+
+    public function createMap(?ProviderInterface $provider = null): Map
+    {
+        $provider ??= $this->defaultProvider;
+        $map = new Map($provider, $provider::getDefaultMapOptions());
+
+        $this->mapRegistry->register($map);
+
+        return $map;
+    }
```

Comment:
I really would like to avoid the registry thing if possible :|  Especially when we do not need it for AM

Reply:
Maybe I am missing something, but I don't see why this is related to the AssetMapper?

Here we register each map in order to render their provider configuration when calling `{{ ux_map_script_tags() }}`

Reply:
I don't think we need ux_map_script_tags ... at least we should _avoid_ it as it won't be easy to do with Turbo, LiveComponent, ajax requests, etc.

Reply:
You know these subjects better than I do, but maybe we can mimic the map registry behavior when calling `render_map()` for the first time per provider?

Reply:
I'll have a look tomorrow!

Reply:
Fixed